### ISSUE TITLE
Cinnamon / Muffin Segfault and fallback mode, unusable desktop

### ISSUE THREAD

**Issue by kkittykait** [linuxmint/cinnamon#9401]
```
 * Cinnamon version (cinnamon --version)
  Mint 20 , Cinnamon 4.6.6

Graphics:  Device-1: NVIDIA GK107M [GeForce GTX 660M] vendor: ASUSTeK driver: nvidia v: 390.138 bus ID: 01:00.0 
           chip ID: 10de:0fd4 
           Display: x11 server: X.Org 1.20.8 driver: nvidia unloaded: fbdev,modesetting,nouveau,vesa 
           resolution: 1920x1080~60Hz 
           OpenGL: renderer: GeForce GTX 660M/PCIe/SSE2 v: 4.6.0 NVIDIA 390.138 direct render: Yes 

 ```

**Issue**
Unable to use Cinnamon Desktop with multiple screens and one inactive

**Steps to reproduce**
Hardware setup.
 1 Laptop
 1 HDMI external screen

  Log in
  Open Display settings
  Set HDMI external to Primary and ACTIVE
  Set Laptop to INACTIVE
  Log out
  Log back in

**Expected behaviour**
Normal desktop to function with one external screen as primary and laptop inactive

**Actual behaviour**
Cinnamon fallback mode, Muffin segfault in logs
To recover, Set all screens to ACTIVE again and log back in, all must be active before logging out
Also difficult to reposition and store screen positions for mouse crossing edge
Also not possible to set laptop screen to INACTIVE without a fight every time
```

**Comment by kkittykait** [comment]
Quite a few segfault entries for libmuffin.so in the kern.log

... kernel: [  178.645213] cinnamon[5474]: segfault at 28 ip 00007fb707b32ae2 sp 00007ffd444e3780 error 6 in libmuffin.so.0.0.0[7fb707afe000+74000]

**Comment by GustavoFloBe** [comment]
Same issue in PC;

2 monitors, 1 connnected using DisplayPort, 1 using HDMI.

If HDMI monitor is disable cinnamon crash and fallbackmode is actived, DP monitor can't be deactivate from Display settings (cause both monitors to blink and both remind active).

Segfault entries for libmuffin.so in kern.log:

Jun 28 19:16:53 vostro kernel: [ 14.677520] cinnamon[1379]: segfault at 28 ip 00007fdcadc23ae2 sp 00007ffc077cc610 error 6 in libmuffin.so.0.0.0[7fdcadbef000+74000]
Jun 29 03:36:06 vostro kernel: [ 113.343761] cinnamon[1779]: segfault at 28 ip 00007f7313607ae2 sp 00007ffd7a22d180 error 6 in libmuffin.so.0.0.0[7f73135d3000+74000]
Jun 29 05:08:53 vostro kernel: [ 14.436159] cinnamon[1384]: segfault at 28 ip 00007f82869a6ae2 sp 00007ffc0bff4c10 error 6 in libmuffin.so.0.0.0[7f8286972000+74000]
Jun 28 19:16:53 vostro kernel: [ 14.677520] cinnamon[1379]: segfault at 28 ip 00007fdcadc23ae2 sp 00007ffc077cc610 error 6 in libmuffin.so.0.0.0[7fdcadbef000+74000]

**Comment by sallywoog** [comment]
I also have this problem.
I also have to do the following

1) Move laptop display to the left of the external screen (I have to do this step or the laptop screen will not be set to inactive)
2) Apply
3) Set laptop screen to inactive

I always have to set every screen to active so I can log in again or I crash at the desktop

**Comment by mtwebster** [comment]
Does this still occur with cinnamon-desktop 4.6.3?

**Comment by sallywoog** [comment]
> Does this still occur with cinnamon-desktop 4.6.3?

It still happens with the version shipped in Mint 20

**Comment by mtwebster** [comment]
@sallywoog @GustavoFloBe what type of graphics card and driver (if any)?

Can anyone provide a trace of this crash? (Check mintreport, 'crash reports', select the cinnamon crash, 'local files' and paste the crash.tar.gz)

**Comment by sallywoog** [comment]
> @sallywoog @GustavoFloBe what type of graphics card and driver (if any)?
> 
> Can anyone provide a trace of this crash? (Check mintreport, 'crash reports', select the cinnamon crash, 'local files' and paste the crash.tar.gz)

cinnamon[2587]: segfault at 28 ip 00007f572da44ae2 sp 00007ffddb697f10 error 6 in libmuffin.so.0.0.0[7f572da10000+74000]

**Comment by mtwebster** [comment]
That is not useful.  I need a full trace.  In Mint, mintreport is the easiest way to do this.  Other distros have other methods.

**Comment by sallywoog** [comment]
> That is not useful. I need a full trace. In Mint, mintreport is the easiest way to do this. Other distros have other methods.

Can you not reproduce the segfault?

**Comment by mtwebster** [comment]
I understand the directions to reproduce.  No I cannot reproduce it.  Why are you so resistant to providing additional information? I'm not trying to do a 'blind fix' I'm trying to gather as much information as possible.

**Comment by sallywoog** [comment]
> I understand the directions to reproduce. No I cannot reproduce it. Why are you so resistant to providing additional information? I'm not trying to do a 'blind fix' I'm trying to gather as much information as possible.

Do you have nvidia drivers installed?

**Comment by mtwebster** [comment]
I've tried with 390 and 440 in both discrete and hybrid mode on my laptop, and cannot reproduce it in any fashion.  The only thing I have not tried is with the modesetting driver.

Distros that use systemd typically use systemd-coredump/coredumpctl.  I suppose for now I'll wait to see if others can provide more.

**Comment by sallywoog** [comment]
> I've tried with 390 and 440 in both discrete and hybrid mode on my laptop, and cannot reproduce it in any fashion. The only thing I have not tried is with the modesetting driver.
> 
> Distros that use systemd typically use systemd-coredump/coredumpctl. I suppose for now I'll wait to see if others can provide more.

I have no core dumps for the kernel and for muffin, possibly because kerneloops is needed?

running  sudo coredumpctrl list shows nothing for it

**Comment by GustavoFloBe** [comment]
@mtwebster My GPU is a Gygabite Gtx 1050 ti

I use the nvidia driver 440 but I've tried to use the other property drivers and even the nouveau and the issue persists.

I uploaded the last crash report to drobox and here's the link. I'm pretty noob in Linux so if that's not what you needed let me know how to send you the right thing or let me know if you are interested in doing remote access.

[https://www.dropbox.com/s/ignv2z5ygb07vki/crash.tar.gz?dl=0](url)

**Comment by mtwebster** [comment]
Thanks, and for the record we would never *dream* of trying to do any sort of remote access, just as I would never allow it myself.  That used to be the first thing I'd disable in windows :)

**Comment by mtwebster** [comment]
@GustavoFloBe can you provide the output from `xrandr --verbose`?

Thanks

**Comment by GustavoFloBe** [comment]
I can not longer reproduce the error, 

I didn't use mint for some days, the only changes I can think of is that now Windows Boost partition is set as primary and if I want to start with mint I need to manually select the partition, and today's updates from update manager.

This is the output you asked for:

`Screen 0: minimum 8 x 8, current 4480 x 1080, maximum 32767 x 32767
DVI-D-0 disconnected (normal left inverted right x axis y axis)
	Identifier: 0x1bd
	Timestamp:  29270
	Subpixel:   unknown
	Clones:    
	CRTCs:      0 1 2 3
	Transform:  1.000000 0.000000 0.000000
	            0.000000 1.000000 0.000000
	            0.000000 0.000000 1.000000
	           filter: 
	CscMatrix: 65536 0 0 0 0 65536 0 0 0 0 65536 0 
	BorderDimensions: 4 
		supported: 4
	Border: 0 0 0 0 
		range: (0, 65535)
	SignalFormat: TMDS 
		supported: TMDS
	ConnectorType: DVI-D 
	ConnectorNumber: 0 
	_ConnectorLocation: 0 
	non-desktop: 0 
		supported: 0, 1
HDMI-0 connected primary 2560x1080+0+0 (0x1bf) normal (normal left inverted right x axis y axis) 673mm x 284mm
	Identifier: 0x1be
	Timestamp:  29270
	Subpixel:   unknown
	Gamma:      1.0:1.0:1.0
	Brightness: 1.0
	Clones:    
	CRTC:       0
	CRTCs:      0 1 2 3
	Transform:  1.000000 0.000000 0.000000
	            0.000000 1.000000 0.000000
	            0.000000 0.000000 1.000000
	           filter: 
	CscMatrix: 65536 0 0 0 0 65536 0 0 0 0 65536 0 
	EDID: 
		00ffffffffffff001e6df159b9b50400
		0a1b010380431c78eaca95a6554ea126
		0f5054a54b80714f818081c0a9c0b300
		0101010101017e4800e0a0381f404040
		3a00a11c21000018023a801871382d40
		582c4500a11c2100001e000000fc004c
		4720554c545241574944450a000000fd
		00384b1e5a18000a202020202020011e
		02031cf1499004031412051f01132309
		07078301000065030c001000023a8018
		71382d40582c450056512100001e011d
		8018711c1620582c250056512100009e
		011d007251d01e206e28550056512100
		001e8c0ad08a20e02d10103e96005651
		21000018000000ff003731304e545747
		39323636350a000000000000000000af
	BorderDimensions: 4 
		supported: 4
	Border: 0 0 0 0 
		range: (0, 65535)
	SignalFormat: TMDS 
		supported: TMDS
	ConnectorType: HDMI 
	ConnectorNumber: 1 
	_ConnectorLocation: 1 
	non-desktop: 0 
		supported: 0, 1
  2560x1080 (0x1bf) 185.580MHz -HSync -VSync *current +preferred
        h: width  2560 start 2624 end 2688 total 2784 skew    0 clock  66.66KHz
        v: height 1080 start 1083 end 1093 total 1111           clock  60.00Hz
  1920x1080 (0x1c0) 148.500MHz +HSync +VSync
        h: width  1920 start 2008 end 2052 total 2200 skew    0 clock  67.50KHz
        v: height 1080 start 1084 end 1089 total 1125           clock  60.00Hz
  1920x1080 (0x1c1) 148.350MHz +HSync +VSync
        h: width  1920 start 2008 end 2052 total 2200 skew    0 clock  67.43KHz
        v: height 1080 start 1084 end 1089 total 1125           clock  59.94Hz
  1920x1080 (0x1c2) 148.500MHz +HSync +VSync
        h: width  1920 start 2448 end 2492 total 2640 skew    0 clock  56.25KHz
        v: height 1080 start 1084 end 1089 total 1125           clock  50.00Hz
  1920x1080 (0x1c3) 74.250MHz +HSync +VSync Interlace
        h: width  1920 start 2008 end 2052 total 2200 skew    0 clock  33.75KHz
        v: height 1080 start 1084 end 1094 total 1124           clock  60.05Hz
  1920x1080 (0x1c4) 74.180MHz +HSync +VSync Interlace
        h: width  1920 start 2008 end 2052 total 2200 skew    0 clock  33.72KHz
        v: height 1080 start 1084 end 1094 total 1124           clock  60.00Hz
  1920x1080 (0x1c5) 74.250MHz +HSync +VSync Interlace
        h: width  1920 start 2448 end 2492 total 2640 skew    0 clock  28.12KHz
        v: height 1080 start 1084 end 1094 total 1124           clock  50.04Hz
  1680x1050 (0x1c6) 146.250MHz -HSync +VSync
        h: width  1680 start 1784 end 1960 total 2240 skew    0 clock  65.29KHz
        v: height 1050 start 1053 end 1059 total 1089           clock  59.95Hz
  1600x900 (0x1c7) 108.000MHz +HSync +VSync
        h: width  1600 start 1624 end 1704 total 1800 skew    0 clock  60.00KHz
        v: height  900 start  901 end  904 total 1000           clock  60.00Hz
  1280x1024 (0x1c8) 135.000MHz +HSync +VSync
        h: width  1280 start 1296 end 1440 total 1688 skew    0 clock  79.98KHz
        v: height 1024 start 1025 end 1028 total 1066           clock  75.02Hz
  1280x1024 (0x1c9) 108.000MHz +HSync +VSync
        h: width  1280 start 1328 end 1440 total 1688 skew    0 clock  63.98KHz
        v: height 1024 start 1025 end 1028 total 1066           clock  60.02Hz
  1280x720 (0x1ca) 74.250MHz +HSync +VSync
        h: width  1280 start 1390 end 1430 total 1650 skew    0 clock  45.00KHz
        v: height  720 start  725 end  730 total  750           clock  60.00Hz
  1280x720 (0x1cb) 74.180MHz +HSync +VSync
        h: width  1280 start 1390 end 1430 total 1650 skew    0 clock  44.96KHz
        v: height  720 start  725 end  730 total  750           clock  59.94Hz
  1280x720 (0x1cc) 74.250MHz +HSync +VSync
        h: width  1280 start 1720 end 1760 total 1980 skew    0 clock  37.50KHz
        v: height  720 start  725 end  730 total  750           clock  50.00Hz
  1152x864 (0x1cd) 108.000MHz +HSync +VSync
        h: width  1152 start 1216 end 1344 total 1600 skew    0 clock  67.50KHz
        v: height  864 start  865 end  868 total  900           clock  75.00Hz
  1024x768 (0x1ce) 78.750MHz +HSync +VSync
        h: width  1024 start 1040 end 1136 total 1312 skew    0 clock  60.02KHz
        v: height  768 start  769 end  772 total  800           clock  75.03Hz
  1024x768 (0x1cf) 65.000MHz -HSync -VSync
        h: width  1024 start 1048 end 1184 total 1344 skew    0 clock  48.36KHz
        v: height  768 start  771 end  777 total  806           clock  60.00Hz
  800x600 (0x1d0) 49.500MHz +HSync +VSync
        h: width   800 start  816 end  896 total 1056 skew    0 clock  46.88KHz
        v: height  600 start  601 end  604 total  625           clock  75.00Hz
  800x600 (0x1d1) 40.000MHz +HSync +VSync
        h: width   800 start  840 end  968 total 1056 skew    0 clock  37.88KHz
        v: height  600 start  601 end  605 total  628           clock  60.32Hz
  720x576 (0x1d2) 27.000MHz -HSync -VSync
        h: width   720 start  732 end  796 total  864 skew    0 clock  31.25KHz
        v: height  576 start  581 end  586 total  625           clock  50.00Hz
  720x480 (0x1d3) 27.000MHz -HSync -VSync
        h: width   720 start  736 end  798 total  858 skew    0 clock  31.47KHz
        v: height  480 start  489 end  495 total  525           clock  59.94Hz
  640x480 (0x1d4) 31.500MHz -HSync -VSync
        h: width   640 start  656 end  720 total  840 skew    0 clock  37.50KHz
        v: height  480 start  481 end  484 total  500           clock  75.00Hz
  640x480 (0x1d5) 25.175MHz -HSync -VSync
        h: width   640 start  656 end  752 total  800 skew    0 clock  31.47KHz
        v: height  480 start  490 end  492 total  525           clock  59.94Hz
  640x480 (0x1d6) 25.170MHz -HSync -VSync
        h: width   640 start  656 end  752 total  800 skew    0 clock  31.46KHz
        v: height  480 start  490 end  492 total  525           clock  59.93Hz
DP-0 connected 1920x1080+2560+0 (0x1c0) normal (normal left inverted right x axis y axis) 598mm x 336mm
	Identifier: 0x1d7
	Timestamp:  29270
	Subpixel:   unknown
	Gamma:      1.0:1.0:1.0
	Brightness: 1.0
	Clones:    
	CRTC:       1
	CRTCs:      0 1 2 3
	Transform:  1.000000 0.000000 0.000000
	            0.000000 1.000000 0.000000
	            0.000000 0.000000 1.000000
	           filter: 
	CscMatrix: 65536 0 0 0 0 65536 0 0 0 0 65536 0 
	EDID: 
		00ffffffffffff00220e803501010101
		1f1c0104a53c22783b0b75a655539f28
		0e5054a54b00d1c0a9c081c0b3009500
		8100d1d9d1e8023a801871382d40582c
		450056502100001e000000fd003090a7
		a723010a202020202020000000fc0048
		50203237780a202020202020000000ff
		00434e4b383331305a5a510a202001a2
		020316f149901f041303120211012309
		070783010000023a801871382d40582c
		450056502100001e023a80d072382d40
		102c458056502100001e866f80a07038
		40403020350056502100001a5a8780a0
		70384d403020350056502100001a0000
		00000000000000000000000000000000
		0000000000000000000000000000009e
	BorderDimensions: 4 
		supported: 4
	Border: 0 0 0 0 
		range: (0, 65535)
	SignalFormat: DisplayPort 
		supported: DisplayPort
	ConnectorType: DisplayPort 
	ConnectorNumber: 2 
	_ConnectorLocation: 2 
	non-desktop: 0 
		supported: 0, 1
  1920x1080 (0x1c0) 148.500MHz +HSync +VSync *current +preferred
        h: width  1920 start 2008 end 2052 total 2200 skew    0 clock  67.50KHz
        v: height 1080 start 1084 end 1089 total 1125           clock  60.00Hz
  1920x1080 (0x1d8) 346.500MHz +HSync -VSync
        h: width  1920 start 1968 end 2000 total 2080 skew    0 clock 166.59KHz
        v: height 1080 start 1083 end 1088 total 1157           clock 143.98Hz
  1920x1080 (0x1d9) 285.500MHz +HSync -VSync
        h: width  1920 start 1968 end 2000 total 2080 skew    0 clock 137.26KHz
        v: height 1080 start 1083 end 1088 total 1144           clock 119.98Hz
  1920x1080 (0x1da) 302.500MHz -HSync +VSync
        h: width  1920 start 2072 end 2280 total 2640 skew    0 clock 114.58KHz
        v: height 1080 start 1083 end 1088 total 1147           clock  99.90Hz
  1920x1080 (0x1db) 253.250MHz -HSync +VSync
        h: width  1920 start 2064 end 2272 total 2624 skew    0 clock  96.51KHz
        v: height 1080 start 1083 end 1088 total 1137           clock  84.88Hz
  1920x1080 (0x1c1) 148.350MHz +HSync +VSync
        h: width  1920 start 2008 end 2052 total 2200 skew    0 clock  67.43KHz
        v: height 1080 start 1084 end 1089 total 1125           clock  59.94Hz
  1920x1080 (0x1c2) 148.500MHz +HSync +VSync
        h: width  1920 start 2448 end 2492 total 2640 skew    0 clock  56.25KHz
        v: height 1080 start 1084 end 1089 total 1125           clock  50.00Hz
  1680x1050 (0x1c6) 146.250MHz -HSync +VSync
        h: width  1680 start 1784 end 1960 total 2240 skew    0 clock  65.29KHz
        v: height 1050 start 1053 end 1059 total 1089           clock  59.95Hz
  1600x900 (0x1c7) 108.000MHz +HSync +VSync
        h: width  1600 start 1624 end 1704 total 1800 skew    0 clock  60.00KHz
        v: height  900 start  901 end  904 total 1000           clock  60.00Hz
  1440x900 (0x1dc) 106.500MHz -HSync +VSync
        h: width  1440 start 1520 end 1672 total 1904 skew    0 clock  55.93KHz
        v: height  900 start  903 end  909 total  934           clock  59.89Hz
  1280x1024 (0x1c8) 135.000MHz +HSync +VSync
        h: width  1280 start 1296 end 1440 total 1688 skew    0 clock  79.98KHz
        v: height 1024 start 1025 end 1028 total 1066           clock  75.02Hz
  1280x800 (0x1dd) 83.500MHz -HSync +VSync
        h: width  1280 start 1352 end 1480 total 1680 skew    0 clock  49.70KHz
        v: height  800 start  803 end  809 total  831           clock  59.81Hz
  1280x720 (0x1ca) 74.250MHz +HSync +VSync
        h: width  1280 start 1390 end 1430 total 1650 skew    0 clock  45.00KHz
        v: height  720 start  725 end  730 total  750           clock  60.00Hz
  1280x720 (0x1cb) 74.180MHz +HSync +VSync
        h: width  1280 start 1390 end 1430 total 1650 skew    0 clock  44.96KHz
        v: height  720 start  725 end  730 total  750           clock  59.94Hz
  1280x720 (0x1cc) 74.250MHz +HSync +VSync
        h: width  1280 start 1720 end 1760 total 1980 skew    0 clock  37.50KHz
        v: height  720 start  725 end  730 total  750           clock  50.00Hz
  1024x768 (0x1ce) 78.750MHz +HSync +VSync
        h: width  1024 start 1040 end 1136 total 1312 skew    0 clock  60.02KHz
        v: height  768 start  769 end  772 total  800           clock  75.03Hz
  1024x768 (0x1cf) 65.000MHz -HSync -VSync
        h: width  1024 start 1048 end 1184 total 1344 skew    0 clock  48.36KHz
        v: height  768 start  771 end  777 total  806           clock  60.00Hz
  800x600 (0x1d0) 49.500MHz +HSync +VSync
        h: width   800 start  816 end  896 total 1056 skew    0 clock  46.88KHz
        v: height  600 start  601 end  604 total  625           clock  75.00Hz
  800x600 (0x1d1) 40.000MHz +HSync +VSync
        h: width   800 start  840 end  968 total 1056 skew    0 clock  37.88KHz
        v: height  600 start  601 end  605 total  628           clock  60.32Hz
  720x576 (0x1d2) 27.000MHz -HSync -VSync
        h: width   720 start  732 end  796 total  864 skew    0 clock  31.25KHz
        v: height  576 start  581 end  586 total  625           clock  50.00Hz
  720x480 (0x1d3) 27.000MHz -HSync -VSync
        h: width   720 start  736 end  798 total  858 skew    0 clock  31.47KHz
        v: height  480 start  489 end  495 total  525           clock  59.94Hz
  640x480 (0x1d4) 31.500MHz -HSync -VSync
        h: width   640 start  656 end  720 total  840 skew    0 clock  37.50KHz
        v: height  480 start  481 end  484 total  500           clock  75.00Hz
  640x480 (0x1d5) 25.175MHz -HSync -VSync
        h: width   640 start  656 end  752 total  800 skew    0 clock  31.47KHz
        v: height  480 start  490 end  492 total  525           clock  59.94Hz
  640x480 (0x1d6) 25.170MHz -HSync -VSync
        h: width   640 start  656 end  752 total  800 skew    0 clock  31.46KHz
        v: height  480 start  490 end  492 total  525           clock  59.93Hz
DP-1 disconnected (normal left inverted right x axis y axis)
	Identifier: 0x1de
	Timestamp:  29270
	Subpixel:   unknown
	Clones:    
	CRTCs:      0 1 2 3
	Transform:  1.000000 0.000000 0.000000
	            0.000000 1.000000 0.000000
	            0.000000 0.000000 1.000000
	           filter: 
	CscMatrix: 65536 0 0 0 0 65536 0 0 0 0 65536 0 
	BorderDimensions: 4 
		supported: 4
	Border: 0 0 0 0 
		range: (0, 65535)
	SignalFormat: TMDS 
		supported: TMDS
	ConnectorType: DisplayPort 
	ConnectorNumber: 2 
	_ConnectorLocation: 2 
	non-desktop: 0 
		supported: 0, 1
`

**Comment by mtwebster** [comment]
@sallywoog are you by chance using a non-default refresh rate for your monitors (not marked "Recommended" in display settings)?

**Comment by sallywoog** [comment]
> @sallywoog are you by chance using a non-default refresh rate for your monitors (not marked "Recommended" in display settings)?

I use the listed refresh rates (recommended) I only use native resolutions when available

**Comment by mtwebster** [comment]
Ok this should be fixed here:
https://github.com/linuxmint/muffin/commit/c092579b67febb7e2a49361942f20f78c82f968e

I finally managed to reproduce the crash.
There are packages for mint 20 if anyone wants to try:
https://github.com/linuxmint/muffin/releases/tag/master.mint20

You'd probably just need the gir1.2-, libmuffin0 and muffin-common packages.  The rest may want dependencies you don't have enabled.  Regardless, I'll try to get a release pushed out asap.

**Comment by ItzSwirlz** [comment]
Glad you found the patch. I'll make sure I fix it through Bullseye later tomorrow.

**Comment by ItzSwirlz** [comment]
In the meantime-can we close this issue and mark it as resolved?

**Comment by mtwebster** [comment]
Closing as this should be fixed as of muffin 4.6.3 (and backported to 4.4.4)
